packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -156,38 +180,91 @@ export class HttpRoute extends Resource implements IHttpRoute {
       target: `integrations/${integration.integrationId}`,
-      authorizerId: authBindResult?.authorizerId,
-      authorizationType: authBindResult?.authorizationType ?? 'NONE', // must be explicitly NONE (not undefined) for stack updates to work correctly
-      authorizationScopes,
+      authorizerId: Lazy.string({ produce: () => this.authBindResult?.authorizerId }),
+      authorizationType: Lazy.string({
+        produce: () => {
+          // Provide the authorization type or explicitly 'NONE'. We must use
+          // 'NONE' and not undefined or CloudFormation doesn't remove an
+          // authorizer. @see https://github.com/aws/aws-cdk/pull/14424
+          return this.authBindResult?.authorizationType ?? 'NONE';
+        },
+      }),
+      authorizationScopes: Lazy.list({
+        produce: () => {
+          let authorizationScopes = this.authBindResult?.authorizationScopes;
+
+          if (this.authBindResult && props.authorizationScopes) {
+            authorizationScopes = Array.from(new Set([
+              ...authorizationScopes ?? [],
+              ...props.authorizationScopes,
+            ]));
+          }
+
+          if (authorizationScopes?.length === 0) {
+            authorizationScopes = undefined;
+          }
+
+          return authorizationScopes;
+        },
+      }),
     };
 
     const route = new CfnRoute(this, 'Resource', routeProps);
     this.routeId = route.ref;
   }
+
+  protected onPrepare() {
+    super.onPrepare();
+    this.tryAuthorizerBinding();
+  }
+
+  private tryAuthorizerBinding() {
+    if (this.authorizer && !this.authBindResult) {
+      this.authBindResult = this.authorizer.bind({
+        route: this,
+        scope: this.httpApi instanceof Construct ? this.httpApi : this, // scope under the API if it's not imported
+      });
+
+      if (!(this.authBindResult.authorizationType in HttpRouteAuthorizationType)) {
+        throw new Error('authorizationType should either be AWS_IAM, JWT, CUSTOM, or NONE');
+      }
+    }
+  }
+
+  public grantInvoke(grantee: iam.IGrantable, options: GrantInvokeOptions = {}): iam.Grant {
+    if (this.authorizer && !(this.authorizer instanceof HttpIamAuthorizer)) {
+      throw new Error('The authorizer has been set, so we cannot enable IAM authorization');
+    }
+
+    if (!this.authorizer) {
+      this.authorizer = new HttpIamAuthorizer();
+    }
+
+    const path = this.path ?? '/';
+    const stage = '*';
+    // When the user has provided a path with path variables, we replace the
+    // path variable and the rest of the path with a wildcard.
+    const iamPath = path.replace(/\{.*?\}.*/, '*');
+
+    const resourceArns = (options.httpMethods ?? [HttpMethod.ANY]).map(httpMethod => {
+      const iamHttpMethod = httpMethod === HttpMethod.ANY ? '*' : httpMethod;
+      const resourceArn = `arn:aws:execute-api:${this.stack.region}:${this.stack.account}:${this.httpApi.apiId}/${stage}/${iamHttpMethod}${iamPath}`;
```

Comment:
I think it would be helpful to make this an accessible attribute on `HttpRoute` like https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigateway.Method.html#methodarn.
That helps constructing policies for accessing the API later on.

Reply:
I've now added `routeArn`.